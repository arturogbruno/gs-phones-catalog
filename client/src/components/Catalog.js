import React from "react";
import ProductOverview from "./ProductOverview";
import CircleLoader from "react-spinners/CircleLoader";

const Catalog = ({ category, products }) => {
    return (
        <div className="Catalog">
            {products.length !== 0 ? (
                <>
                    <h1>{`This is the ${category} catalog`}</h1>
                    {products.map(product => (
                        <ProductOverview key={product.id} category={category}Â product={product} />
                    ))}
                </>
            ) : (
                <CircleLoader 
                    color={"#00b091"}
                    size={50}
                />
            )}
        </div>
    )
}

export default Catalog;