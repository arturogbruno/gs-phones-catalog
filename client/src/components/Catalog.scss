.Catalog {
    width: 60%;
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-bottom: 50px;
}