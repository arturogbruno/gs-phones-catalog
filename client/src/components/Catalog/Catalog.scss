@import "../../styles/mixins.scss";

.Catalog {
    width: 80%;
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-bottom: 50px;

    header {
        color: rgb(35, 35, 35);
        display: flex;
        flex-direction: column;
        align-items: center;
        margin-bottom: 3rem;
        text-shadow: 2px 2px 3px rgba(255, 255, 255, 0.8);
        letter-spacing: 1px;
        
        h1 {
            margin-bottom: 1rem;
        }
        
        h3 {
            font-style: italic;
            font-weight: 600;
        }
    }

    @include lg {
        width: 50%;
    }
}