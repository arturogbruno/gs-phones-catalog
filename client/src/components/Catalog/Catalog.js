import React from "react";
import ProductOverview from "../ProductOverview/ProductOverview";
import CircleLoader from "react-spinners/CircleLoader";
import "./Catalog.scss";

const Catalog = ({ category, products }) => {
    return (
        <div className="Catalog">
            {products.length !== 0 ? (
                <>
                    <header>
                        <h1>{`Welcome to the smallest ${category} catalog`}</h1>
                        <h3>We have narrowed the search for you to make your choice easier!</h3>
                    </header>
                    {products.map(product => (
                        <ProductOverview key={product.id} category={category}Â product={product} />
                    ))}
                </>
            ) : (
                <CircleLoader 
                    color={"#fff"}
                    size={50}
                />
            )}
        </div>
    )
}

export default Catalog;