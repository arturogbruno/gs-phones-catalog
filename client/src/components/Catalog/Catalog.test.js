import React from "react";
import { BrowserRouter } from "react-router-dom";
import { mount } from "enzyme";
import Catalog from "./Catalog";

describe("testing Catalog component", () => {
    const mockCategory = "phones";
    let mockProducts = [
        {
            "id": 2,
            "name": "Honor 10",
            "manufacturer": "Huawei",
            "description": "Great phone with an excellent interface. One of the best mid price range phones in the market.",
            "color": "blue",
            "price": 399,
            "imageFileName": "Honor_10.png",
            "screen": "5,84 inch Full-HD",
            "processor": "Kirin 970",
            "ram": 6
        }
    ];

    let wrapper = mount(
        <BrowserRouter>
            <Catalog category={mockCategory} products={mockProducts} />
        </BrowserRouter>
    );

    it("should render Catalog component", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("should render ProductOverview elements if it receives a non-empty products array", () => {
        expect(wrapper.find("ProductOverview").length).toBeGreaterThan(0);
    });
});