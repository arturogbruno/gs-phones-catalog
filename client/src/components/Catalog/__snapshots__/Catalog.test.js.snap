// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Catalog component should render Catalog component 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Catalog
      category="phones"
      products={
        Array [
          Object {
            "color": "blue",
            "description": "Great phone with an excellent interface. One of the best mid price range phones in the market.",
            "id": 2,
            "imageFileName": "Honor_10.png",
            "manufacturer": "Huawei",
            "name": "Honor 10",
            "price": 399,
            "processor": "Kirin 970",
            "ram": 6,
            "screen": "5,84 inch Full-HD",
          },
        ]
      }
    >
      <div
        className="Catalog"
      >
        <header>
          <h1>
            Welcome to the smallest phones catalog
          </h1>
          <h3>
            We have narrowed the search for you to make your choice easier!
          </h3>
        </header>
        <ProductOverview
          category="phones"
          key="2"
          product={
            Object {
              "color": "blue",
              "description": "Great phone with an excellent interface. One of the best mid price range phones in the market.",
              "id": 2,
              "imageFileName": "Honor_10.png",
              "manufacturer": "Huawei",
              "name": "Honor 10",
              "price": 399,
              "processor": "Kirin 970",
              "ram": 6,
              "screen": "5,84 inch Full-HD",
            }
          }
        >
          <div
            className="ProductOverview"
          >
            <Link
              to="/phones/2"
            >
              <LinkAnchor
                href="/phones/2"
                navigate={[Function]}
              >
                <a
                  href="/phones/2"
                  onClick={[Function]}
                >
                  <img
                    alt="Huawei Honor 10"
                    src="images/Honor_10.png"
                  />
                  <div
                    className="productMainInfo"
                  >
                    <div
                      className="makerModel"
                    >
                      <h2>
                        Huawei
                      </h2>
                      <h2>
                        Honor 10
                      </h2>
                    </div>
                    <h2
                      className="price"
                    >
                      399
                       â‚¬
                    </h2>
                  </div>
                </a>
              </LinkAnchor>
            </Link>
          </div>
        </ProductOverview>
      </div>
    </Catalog>
  </Router>
</BrowserRouter>
`;
